### PR TITLE
Documentation configuration files

### LINKED ISSUE #3375: document that not using the canonical configfile may crash other tools
followup to pypa/pip#5182

iniconfig is not necessarily compatible with what other tools expect, at least for setup.cfg config options of pytest can crash other tools

### PR DESCRIPTION
**Pull request by yoavcaspi**
Fix #3375

### CODE REVIEW

File: doc/en/reference.rst [hunk 1]
```diff
@@ -889,6 +889,9 @@ Here is a list of builtin configuration options that may be written in a ``pytes
 file, usually located at the root of your repository. All options must be under a ``[pytest]`` section
 (``[tool:pytest]`` for ``setup.cfg`` files).
 
+.. warning:: usage of configuration file different than the canonical, i.e. ``pytest.ini``
```
Comment (RonnyPfannschmidt) [inline doc/en/reference.rst]: this warning should be specifically for `setup.cfg` tox and pytest use the same ini parser

Comment (nicoddemus) [reply doc/en/reference.rst]: Adding a new commit or amending and force pushing are all valid. :+1:

File: doc/en/reference.rst [hunk 2]
```diff
@@ -889,6 +889,9 @@ Here is a list of builtin configuration options that may be written in a ``pytes
...
+   can break other tools configurations.
```
Comment (nicoddemus) [inline doc/en/reference.rst]: My suggestion:

```rst
.. warning:: 
	Usage of ``setup.cfg`` is not recommended unless for very simple use cases: ``.cfg``
    files use a different parser than ``pytest.ini`` and ``tox.ini`` which might cause hard to track
    down problems. 
    When possible, it is recommended to use the latter files to hold your pytest configuration.
```

Comment (yoavcaspi) [reply doc/en/reference.rst]: Thanks. I like this suggestion :-)

**Comment by nicoddemus** [comment]
Thanks!

**Comment by nicoddemus** [comment]
Ignoring AppVeyor because codecov is acting up again

**Pull request merged**